torchvision/transforms/functional_tensor.py

```diff
@@ -145,16 +145,19 @@ def crop(img: Tensor, top: int, left: int, height: int, width: int) -> Tensor:
 def rgb_to_grayscale(img: Tensor, num_output_channels: int = 1) -> Tensor:
     if img.ndim < 3:
         raise TypeError(f"Input image tensor should have at least 3 dimensions, but found {img.ndim}")
```

Comment:
Question: What if input is a 2d grayscale image `img.ndim == 2`? Should we unsqueeze and treat it as 1xHxW? We handle this in some limited kernels. Thoughts?

Reply:
Yes, we could support 2d image tensors as we accept 2d tensors as images:
https://github.com/pytorch/vision/blob/fb35989df6e94f5dd0fbbf11027646f521e7bbbe/torchvision/transforms/functional_tensor.py#L9-L10

Reply:
Do we want to add new features to a deprecated functional that are not bug fixes?

Reply:
As discussed offline, we are not going to add support for 2D images here, since we are not sure if they are supported throughout the API in general. Consensus was that we either need to accept them everywhere or only in the "to tensor image" logic.